From 47fef158e4a38c57a0c13349ce829295b2f85277 Mon Sep 17 00:00:00 2001
From: Dennis Hezel <dh@3yourmind.com>
Date: Tue, 11 Jan 2022 17:26:44 +0100
Subject: [PATCH] do-not-link-std-coroutines

---
 source/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 7d118a5..655ca33 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -77,7 +77,7 @@ target_include_directories(unifex
 target_compile_features(unifex PUBLIC cxx_std_17)
 
 if(CXX_COROUTINES_HAVE_COROUTINES)
-  target_link_libraries(unifex PUBLIC std::coroutines)
+  # target_link_libraries(unifex PUBLIC std::coroutines)
 endif()
 
 configure_file(unifex.pc.in unifex.pc @ONLY)
-- 
2.32.0.windows.2

