From a194733d2ff9411291086d52c8bcb66989029996 Mon Sep 17 00:00:00 2001
From: Dennis Hezel <dh@3yourmind.com>
Date: Thu, 11 Nov 2021 19:50:37 +0100
Subject: [PATCH] fix-install

---
 source/CMakeLists.txt | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 3f25517..126556b 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -67,10 +67,23 @@ configure_file(
 target_include_directories(unifex
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include/>
-    "${PROJECT_BINARY_DIR}/include")
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
 
 target_compile_features(unifex PUBLIC cxx_std_17)
 
-if(CXX_COROUTINES_HAVE_COROUTINES)
-  target_link_libraries(unifex PUBLIC std::coroutines)
-endif()
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/unifex
+        DESTINATION include
+)
+install(FILES ${PROJECT_BINARY_DIR}/include/unifex/config.hpp
+        DESTINATION include/unifex
+)
+
+install(TARGETS unifex
+  EXPORT unifex-config
+  DESTINATION lib
+)
+install(EXPORT unifex-config
+  NAMESPACE unofficial::
+  DESTINATION lib/cmake/unifex
+)
-- 
2.25.1

