From 048b9658efa6442c27e0df8e1c132de276c647bc Mon Sep 17 00:00:00 2001
From: Dennis Hezel <dh@3yourmind.com>
Date: Mon, 1 Nov 2021 18:50:40 +0100
Subject: [PATCH] fix await_transform forward declaration

---
 include/unifex/await_transform.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/unifex/await_transform.hpp b/include/unifex/await_transform.hpp
index d895d05..5ae8541 100644
--- a/include/unifex/await_transform.hpp
+++ b/include/unifex/await_transform.hpp
@@ -183,7 +183,7 @@ public:
 template <typename Promise, typename Sender>
 using _as_awaitable = typename _awaitable<Promise, Sender>::type;
 
-inline constexpr struct _fn {
+inline const struct _fn {
   // Call custom implementation if present.
   template(typename Promise, typename Value)
     (requires tag_invocable<_fn, Promise&, Value>)
-- 
2.32.0.windows.2

