diff --git a/include/unifex/spawn_future.hpp b/include/unifex/spawn_future.hpp
index 23c7e33..9d41093 100644
--- a/include/unifex/spawn_future.hpp
+++ b/include/unifex/spawn_future.hpp
@@ -398,13 +398,13 @@ struct _spawn_future_op<T...>::type : _spawn_future_op_base {
   // returns a Sender that produces the values produced by the spawned
   // operation
   auto get_value_sender() noexcept(
-      noexcept(apply(just, std::move(this->values_).get()))) {
+      noexcept(apply(just, std::move(values_).get()))) {
     return apply(just, std::move(values_).get());
   }
 
   // returns a Sender that produces the error produced by the spawned operation
   auto get_error_sender() noexcept(
-      noexcept(just_error(std::move(this->error_).get()))) {
+      noexcept(just_error(std::move(error_).get()))) {
     return just_error(std::move(error_).get());
   }
 
