diff --git a/absl/strings/cord.h b/absl/strings/cord.h
index 1d8fcd37..2e8fd500 100644
--- a/absl/strings/cord.h
+++ b/absl/strings/cord.h
@@ -849,8 +849,8 @@ class Cord {
   friend bool operator==(const Cord& lhs, const Cord& rhs);
   friend bool operator==(const Cord& lhs, absl::string_view rhs);
 
-#ifdef __cpp_impl_three_way_comparison
-
+#if defined(__cpp_impl_three_way_comparison) && \
+    defined(__cpp_lib_three_way_comparison)
   // Cords support comparison with other Cords and string_views via operator<
   // and others; here we provide a wrapper for the C++20 three-way comparison
   // <=> operator.
