diff --git a/include/boost/asio/experimental/impl/promise.hpp b/include/boost/asio/experimental/impl/promise.hpp
index 0cab8475..2dd2efc2 100644
--- a/include/boost/asio/experimental/impl/promise.hpp
+++ b/include/boost/asio/experimental/impl/promise.hpp
@@ -40,6 +40,7 @@ template<typename... Ts, typename Executor, typename Allocator>
 struct promise_impl<void(Ts...), Executor, Allocator>
 {
   using result_type = std::tuple<Ts...>;
+  static constexpr auto ALIGN = alignof(result_type);
 
   promise_impl(Allocator allocator, Executor executor)
     : allocator(std::move(allocator)), executor(std::move(executor))
@@ -58,7 +59,7 @@ struct promise_impl<void(Ts...), Executor, Allocator>
   }
 
   typename aligned_storage<sizeof(result_type),
-    alignof(result_type)>::type result;
+    ALIGN>::type result;
   std::atomic<bool> done{false};
   cancellation_signal cancel;
   Allocator allocator;
