diff --git a/src/google/protobuf/message_lite.h b/src/google/protobuf/message_lite.h
index cae4e6bbe..8b643609b 100644
--- a/src/google/protobuf/message_lite.h
+++ b/src/google/protobuf/message_lite.h
@@ -22,6 +22,9 @@
 #include <iosfwd>
 #include <string>
 #include <type_traits>
+#if defined(__cpp_lib_three_way_comparison)
+#include <compare>
+#endif
 
 #include "absl/base/attributes.h"
 #include "absl/log/absl_check.h"
