diff --git a/src/google/protobuf/message_lite.h b/src/google/protobuf/message_lite.h
index cae4e6bbe..ab5da650c 100644
--- a/src/google/protobuf/message_lite.h
+++ b/src/google/protobuf/message_lite.h
@@ -22,6 +22,9 @@
 #include <iosfwd>
 #include <string>
 #include <type_traits>
+#if defined(__cpp_lib_three_way_comparison)
+#include <compare>
+#endif
 
 #include "absl/base/attributes.h"
 #include "absl/log/absl_check.h"
@@ -907,7 +910,7 @@ class TypeId {
     return a.data_ >= b.data_;
   }
 
-#if defined(__cpp_impl_three_way_comparison) && \
+#if defined (__cpp_lib_three_way_comparison) && defined(__cpp_impl_three_way_comparison) && \
     __cpp_impl_three_way_comparison >= 201907L
   friend constexpr auto operator<=>(TypeId a, TypeId b) {
     return a.data_ <=> b.data_;
