diff --git a/src/google/protobuf/port.h b/src/google/protobuf/port.h
index 182f5786f..a6f292f15 100644
--- a/src/google/protobuf/port.h
+++ b/src/google/protobuf/port.h
@@ -173,10 +173,6 @@ void AssertDownCast(From* from) {
 #if defined(__cpp_concepts)
   // Check that this function is not used to downcast message types.
   // For those we should use {Down,Dynamic}CastTo{Message,Generated}.
-  static_assert(!requires {
-    std::derived_from<std::remove_pointer_t<To>,
-                      typename std::remove_pointer_t<To>::MessageLite>;
-  });
 #endif
 
 #if PROTOBUF_RTTI
